prow/crier/controller.go

```diff
@@ -182,7 +182,27 @@ func (c *Controller) updateReportState(pj *v1.ProwJob) error {
 	logrus.Infof("Created merge patch: %v", string(patch))
 
 	_, err = c.pjclientset.Prow().ProwJobs(pj.Namespace).Patch(pj.Name, types.MergePatchType, patch)
-	return err
+	if err != nil {
+		return err
+	}
+
+	// Block until the update is in the lister to make sure that events from another controller
+	// that also does reporting dont trigger another report because our lister doesn't yet contain
+	// the updated Status
+	if err := wait.Poll(time.Second, 3*time.Second, func() (bool, error) {
```

Comment:
Thanks for catch this! I think I've also see this before but github report behaves as a no-op if report the same failure twice so it's slipped.

how long it normally takes to populate into the lister cache? It might affect performance a bit.

Reply:
We have a proprietary controller that does basically the same, I remember times of 200-300ms. This was some time ago so my memories may be wrong though. Idea with the 1s was to be 80% sure we have the update on the first poll..

I could run this with a smaller poll interval and logging that tells the iteration on which the update was present. Not sure if it's valid to extrapolate from our setup to the K8s one which has roughly 10x as many prowjobs. And at the end of the day if this is a concern, you can just increase the worker count. They cost almost nothing, the go runtime will schedule them off of their thread while they are waiting so they don't block it needlessly.

Reply:
the gerrit reporter currently only can run as a single worker (for reasons) which might be problematic if it goes too slow, but we can see how it goes and tune the polling interval.